### ISSUE TITLE
Find redundant GHA and remove them

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2295]
Lately I've found there's a lot of lethargy in our review process because of queueing in the GHA for the repo.

My suspicion is that many of the actions we run are totally redundant -- they provide signal that is completely identical to at least one other action. That is, they _always_ share failure status with some other action. Such actions can be removed.

I will be extending this script to analyze this numerically:

https://gist.github.com/MichaelChirico/d5a1a0d864864e0d30f24e3eb87afecf

**Comment by IndrajeetPatil** [comment]
In my view, we can remove:

- `R-CMD-check / ubuntu-latest (devel) zh_CN` (this workflow is actually quite flaky and often fails unpredictably)
- `test-package` (redundant if we run `R-CMD-check` with `options(warn = 2)`)
- `R-CMD-check / windows-latest (4.1)` (if this is only for checking with a newer Rtools version, isn't that also done via R-release?)

**Comment by MichaelChirico** [comment]
Hmm GHA API is doing something weird:

```r
runs[, sum(conclusion == "failure"), by = name]
#                          name  V1
# 1:             check-link-rot   0
# 2:                    pkgdown   0
# 3:                       lint   0
# 4:                R-CMD-check 346
# 5:         check-all-examples   0
# 6:               test-package 173
# 7:           R-CMD-check-hard   0
# 8:              test-coverage   0
# 9: pages build and deployment   0
```

I'm fairly confident some of those other actions have failed! Especially we saw check-lint-rot fail when CRAN was down this week.

**Comment by AshesITR** [comment]
The zh_CN run was added to test a non-UTF 8 locale because of some encoding related bug I think.

**Comment by MichaelChirico** [comment]
Yea, I wouldn't remove `zh_CN` -- it has a clear purpose marking it distinct from the other runs. Flakiness can be addressed separately.

**Comment by IndrajeetPatil** [comment]
Gabor had checked this with R-devel after discussion in https://github.com/r-lib/actions/issues/628/, and they had also mentioned not to use the non-UTF8 locale. 

If we choose to retain this, we shouldn't be running this workflow on R-devel, but maybe on R-oldrel. Otherwise, it's just going to fail sporadically.

**Comment by AshesITR** [comment]
Maybe switch to `r-release`?
Dropping non UTF-8 testing seems like a bad trade. We are also working with user files that might have any encoding, so we should test that it's handled correctly.

**Comment by MichaelChirico** [comment]
I haven't noticed this as much lately, or maybe I'm just used to it. I don't think there's anything to do here. The slowest GHA is usually the Windows one, which is kind of expected IME.
